### PR TITLE
Update tooling

### PR DESCRIPTION
**Pull request by djhi**
- Update more dependencies
- Switch to vite
- Fix CLI

- [x] Check the CLI mode is working
- [x] Check the import through bundlers is working
- [x] Check the import inside Node is working

### CODE REVIEW

File: src/introspection/getSchemaFromData.spec.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import {
     GraphQLString,
 } from 'graphql';
 import getSchemaFromData from './getSchemaFromData';
+import e from 'cors';
```
Comment (fzaninotto) [inline src/introspection/getSchemaFromData.spec.js:10]: This looks weird

File: Makefile [hunk 2]
```diff
@@ -10,13 +10,13 @@ install: package.json ## Install dependencies
 	@$(PKG) install
 
 watch: ## continuously compile ES6 files to JS
-	NODE_ENV=development ./node_modules/.bin/webpack --watch
+	@yarn vite build --watch
 
 test: ## Launch unit tests
-	@NODE_ENV=test ./node_modules/.bin/jest
+	@NODE_ENV=test NODE_OPTIONS="$$NODE_OPTIONS --experimental-vm-modules" ./node_modules/.bin/jest
```
Comment (fzaninotto) [inline Makefile:16]: this I don't understand. Would you ind explaining what it does?

File: bin/json-graphql-server.cjs [hunk 3]
```diff
@@ -23,7 +23,7 @@ process.argv.forEach((arg, index) => {
 });
 
 app.use(cors());
-app.use('/', JsonGraphqlServer(data));
+app.use('/', JsonGraphqlServer({ data }).getHandler());
```
Comment (fzaninotto) [inline bin/json-graphql-server.cjs:26]: That's a breaking change, right?

File: vite.config.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { resolve } from 'path';
+import { defineConfig } from 'vite';
+
+import pkg from './package.json';
+
+export default defineConfig({
+    build: {
+        lib: {
+            // Could also be a dictionary or array of multiple entry points
+            entry: resolve(__dirname, 'src/client.js'),
```
Comment (fzaninotto) [inline vite.config.js:10]: We used to provide 2 libraries: one for the client and one for the server. I only see the client one here.

File: bin/json-graphql-server.cjs [hunk 5]
```diff
@@ -3,7 +3,7 @@ require('reify');
 var path = require('path');
 var express = require('express');
 var cors = require('cors');
-var JsonGraphqlServer = require('../lib/json-graphql-server.node.min').default;
+var JsonGraphqlServer = require('../dist/json-graphql-server').default;
```
Comment (fzaninotto) [inline bin/json-graphql-server.cjs:6]: I suspect the README needs an update, too

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (djhi) [reply Makefile:16]: Make jest accept our ESM code: https://jestjs.io/docs/ecmascript-modules

Comment (fzaninotto) [reply Makefile:16]: Does that mean that, in order to use the CLI, our users will have to use the same flag? This would be a BC.

Comment (djhi) [reply Makefile:16]: Nope, as said in the PR desc, I tested the CLI. I did it without any flag on latest Node LTS

File: README.md [hunk 6]
```diff
@@ -448,15 +448,15 @@ Then use the `jsonGraphqlExpress` express middleware:
...
+import jsonGraphqlExpress from 'json-graphql-server/node';
 
 const PORT = 3000;
 const app = express();
 const data = {
     // ... your data
 };
 
-app.use('/graphql', jsonGraphqlExpress.default(data));
+app.use('/graphql', jsonGraphqlExpress(data));
```
Comment (djhi) [inline README.md:459]:
```suggestion
app.use('/graphql', jsonGraphqlExpress.default(data));
```

File: README.md [hunk 7]
```diff
@@ -448,15 +448,15 @@ Then use the `jsonGraphqlExpress` express middleware:
 
 ```js
 import express from 'express';
-import jsonGraphqlExpress from 'json-graphql-server';
+import jsonGraphqlExpress from 'json-graphql-server/node';
```
Comment (adguernier) [inline README.md:451]:
```suggestion
const { default: jsonGraphqlExpress } = require("json-graphql-server/node");
```

Comment (djhi) [reply README.md:451]: I reverted the change, the doc was right initially

File: vite.config.umd.js [hunk 8]
```diff
@@ -0,0 +1,19 @@
+import { resolve } from 'path';
+import { defineConfig } from 'vite';
+
+import pkg from './package.json';
```
[the remark this answers is not in the document]
Comment (djhi) [reply vite.config.umd.js:4]:
```suggestion
```

**Review by adguernier — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (djhi) [reply Makefile:16]: The minimum required node version is 16.

File: bin/json-graphql-server.cjs (hunk 3 above)
Comment (djhi) [reply bin/json-graphql-server.cjs]: Reverted

**Review by erwanMarmelab — APPROVED** [review]

**Pull request merged**